### DISCUSSION TITLE
Switching between real image and Color Test Image

### DISCUSSION THREAD

**Discussion by JulianBvW** [awawa-dev/HyperHDR#268] [Q&A] [answered]
Hello,

I installed my setup following the tutorial from [Everything Smart Home](https://www.youtube.com/watch?v=J26oYlKyq7Q&t=853s) and everything works fine with one exception:

![image](https://user-images.githubusercontent.com/45465429/167243693-40ca3486-a883-47f9-ba2a-dcff33a483df.png)

Every now and then the TV "disconnects" showing "no signal input" (as if i pulled the HDMI cable) and the LEDs also show something different. After a few seconds everything works fine again until the next signal loss.

Inspecting the LED Visualization reveals that everytime this disconnecton accours, the input for HyperHDR is the static color test image which causes the LEDs to shine according to that image.

![image](https://user-images.githubusercontent.com/45465429/167243394-d5a1cb7e-806d-44e8-bde5-a71755fc38a0.png)

While powering the HDMI Capture Card directly from a power plug this disconnection does not accour (but then there is obviously no input to the Raspberry Pi).

I hope someone can help me figure out what to do here.

Hardware:
- Raspberry Pi 3 B v1.2
- 4k HDR USB 3.0 Game Capture Card HDMI
- WS2812B 5050 RGB LED Strip 5m 150 LEDs
- 10A, 5V DC
-> The Pi and the LED Strip are powered by this one source using a DC barrel jack connector and a cut Micro-USB cable

**Comment by awawa-dev** [comment, marked as answer]
Hi
![link](https://user-images.githubusercontent.com/45465429/167243394-d5a1cb7e-806d-44e8-bde5-a71755fc38a0.png)
This is the 'no-signal' pattern send by your USB grabber. May happen due to:  
- unrecognized or unsupported video signal
- bad HDMI cables
- problems with the USB power supply causing unpredictable USB grabber behavior  

LED power supply is not good for supplying Raspberry Pi and I warned about it in my blog. Period. May work or may cause hard to diagnose problems with Rpi or USB peripherals. If you want to try a separate  USB power supply for Rpi (at least 2A, solid USB power cable is also necessary) then you must also provide the common ground for your systems. Otherwise LED strip may not work, will be blinking or other bad things may happen.

**Reply by JulianBvW** [reply]
Thanks for the quick response!

If I want to connect the Pi and the LED via different power supplies, can I just put a cable from the minus pole of the LEDs to the Ground Pin from the Pi to provide common Ground? Or is it more complicated?

**Reply by awawa-dev** [reply]
Should be OK. BTW on the LED strip it's usually labeled as GND but it's connected to the minus pole of the LED power supply. It should be connected to the RPi ground to provide the common ground. 
Something like that:  
https://github.com/awawa-dev/HyperHDR/discussions/191#discussioncomment-2245758
That's the proper solution including the ESP and the level shifter but it also utilizes common ground from the minus pole of the LED power supply.

**Reply by JulianBvW** [reply]
Everything works fine now, thank yo really much.
